packages/xrpl/src/client/index.ts

```diff
@@ -666,6 +668,33 @@ class Client extends EventEmitter<EventTypes> {
       promises.push(checkAccountDeleteBlockers(this, tx))
     }
 
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment -- ignore type-assertions on the DeliverMax property
+    // @ts-expect-error -- DeliverMax property exists only at the RPC level, not at the protocol level
+    if (tx.TransactionType === 'Payment' && tx.DeliverMax != null) {
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- This is a valid null check for Amount
+      if (tx.Amount == null) {
+        // If only DeliverMax is provided, use it to populate the Amount field
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment -- ignore type-assertions on the DeliverMax property
+        // @ts-expect-error -- DeliverMax property exists only at the RPC level, not at the protocol level
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment -- DeliverMax is a known RPC-level property
+        tx.Amount = tx.DeliverMax
```

Comment:
This wouldn't work for API v1. There should also be a check for API version here.

Reply:
I believe we can use the `DeliverMax` field even with API version 1 machines. Please check the section under "Additions in 2.0" in the API-Changelog: https://github.com/XRPLF/rippled/blob/develop/API-CHANGELOG.md

![image](https://github.com/XRPLF/xrpl.js/assets/21219765/df8220f3-6689-4e5b-bab4-8693d2a038ef)

Reply:
Yes, you're right. From the API-Changelog: "To ease the transition, `DeliverMax` is present regardless of API version, since adding a field is non-breaking."